course_discovery/apps/course_metadata/data_loaders/api.py

```diff
@@ -899,3 +899,99 @@ def _update_program_banner_image(self, body, program):
             program.save()
         else:
             logger.exception('Loading the banner image %s for program %s failed', image_url, program.title)
+
+
+class WordPressApiDataLoader(AbstractDataLoader):
+    """
+    Sync courses data from WordPress.
+    """
+
+    def ingest(self):
+        """
+        Load courses data from the WordPress.
+        """
+        logger.info('Refreshing Courses Data from WordPress %s...', self.partner.marketing_site_api_url)
+        initial_page = 1
+        response = self._make_request(initial_page)
+        count = response['pagination']['count']
+        pages = response['pagination']['num_pages']
+        self._process_response(response)
+
+        pagerange = range(initial_page + 1, pages + 1)
+        logger.info('Looping to request all %d WordPress pages...', pages)
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=self.max_workers) as executor:
+            if self.is_threadsafe:
+                for page in pagerange:
+                    time.sleep(30)
+                    executor.submit(self._load_data, page)
+            else:
+                for future in [executor.submit(self._make_request, page) for page in pagerange]:
+                    response = future.result()
+                    self._process_response(response)
+
+        logger.info('Retrieved %d course from %s.', count, self.partner.marketing_site_api_url)
+
+    def _make_request(self, page):
+        """
+        Send request to WordPress.
+        """
+        logger.info('Requesting WordPress course run page %d...', page)
+        params = {'page': page, 'page_size': self.PAGE_SIZE}
+        auth = requests.auth.HTTPBasicAuth(settings.WORDPRESS_USERNAME, settings.WORDPRESS_APP_PASSWORD)
+        response = requests.get(self.api_url, auth=auth, params=params)
+        response.raise_for_status()
+        return response.json()
+
+    def _load_data(self, page):
+        """
+        Make a request for the given page and process the response.
+        """
+        response = self._make_request(page)
+        self._process_response(response)
+
+    def _process_response(self, response):
+        """
+        Process the response from the WordPress.
+        """
+        results = response['results']
+        logger.info('Retrieved %d WordPress course runs...', len(results))
+
+        for body in results:
+
+            course_run_key = body['course_id']
+            try:
+                body = self.clean_strings(body)
+                course_run = CourseRun.objects.get(key__iexact=course_run_key)
+                course_run.short_description_override = body['excerpt']
+                course_run.full_description_override = body['description']
+                course_run.featured = body['featured']
+                course_run.card_image_url = body['featured_image_url']
+                course_run.slug = body['slug']
+                category = body['category']
```

Comment:
@burhandodhy a course run can have multiple categories as tags. so why we are not treating categories same as tags?

Reply:
@ziafazal admin can only select one category in WordPress as we converted the checkboxes into radio 

![image](https://user-images.githubusercontent.com/7139602/106770191-8a1cc680-665f-11eb-927d-52e2df8e6677.png)

Reply:
@burhandodhy I think that should be checkboxes instead of radio button to allow association of multiple categories with course.

Reply:
Sure. Going to update the code and also create a ticket to add the multiple category support in WordPress.

Reply:
@ziafazal multiple categories support added.